File: Ejercicios POO/Javascript/POO/proveedores.js
// Written By Ismael Heredia in the year 2016

function Proveedor(id_proveedor,nombre_empresa,direccion,telefono,fecha_registro){
	this.id_proveedor=id_proveedor;
	this.nombre_empresa=nombre_empresa;
	this.direccion = direccion;
	this.telefono = telefono;
	this.fecha_registro = fecha_registro;
}

Proveedor.prototype.getId_proveedor=function(){
	return this.id_proveedor;
};
Proveedor.prototype.setId_proveedor=function(id_proveedor){
	this.id_proveedor = id_proveedor;
};

Proveedor.prototype.getNombre_empresa=function(){
	return this.nombre_empresa;
};
Proveedor.prototype.setNombre_empresa=function(nombre_empresa){
	this.nombre_empresa = nombre_empresa;
};

Proveedor.prototype.getDireccion=function(){
	return this.direccion;
};
Proveedor.prototype.setDireccion=function(direccion){
	this.direccion = direccion;
};

Proveedor.prototype.getTelefono=function(){
	return this.telefono;
};
Proveedor.prototype.setTelefono=function(telefono){
	this.telefono = telefono;
};

Proveedor.prototype.getFecha_registro=function(){
	return this.fecha_registro;
};
Proveedor.prototype.setFecha_registro=function(fecha_registro){
	this.fecha_registro = fecha_registro;
};

Proveedor.prototype.toString=function(){
	return "[+] ID Proveedor : "+this.id_proveedor + "\n" + "[+] Nombre empresa : "+this.nombre_empresa
			+ "\n" + "[+] Direccion : "+this.direccion + "\n" + "[+] Telefono : "+this.telefono
			+ "\n" + "[+] Fecha registro : "+this.fecha_registro;
};

File: CRUD/Python/Sistema con SQLServer/Proveedores.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# Written By Ismael Heredia in the year 2017

class Proveedor(object):

   def __init__(self, id="", nombre="", direccion="",telefono="",fecha_registro=""):
       self.__id = id
       self.__nombre = nombre
       self.__direccion = direccion
       self.__telefono = telefono
       self.__fecha_registro = fecha_registro
              
   def getId(self):
       return self.__id

   def setId(self, id = ""):
       self.__id = id

   def getNombre(self):
       return self.__nombre

   def setNombre(self, nombre = ""):
       self.__nombre = nombre

   def getDireccion(self):
       return self.__direccion

   def setDireccion(self, direccion = ""):
       self.__direccion = direccion
    
   def getTelefono(self):
       return self.__telefono

   def setTelefono(self, telefono = ""):
       self.__telefono = telefono
       
   def getFecha_registro(self):
       return self.__fecha_registro

   def setFecha_registro(self, fecha_registro = ""):
       self.__fecha_registro = fecha_registro
       
   def toString(self):
       return "Proveedor{id=%d, nombre=%s, direccion=%s, telefono=%d, fecha_registro=%s}" % (self.__id,self.__nombre,self.__direccion,self.__telefono,self.__fecha_registro)

   def destroy(self):
       self.__id = None
       self.__nombre = None
       self.__direccion = None
       self.__telefono = None
       self.__fecha_registro = None
